### DISCUSSION TITLE
Prolong expiration date encrypted code

### DISCUSSION THREAD

**Discussion by JavierFrade** [dashingsoft/pyarmor#1638] [General]
With a Pyarmor Group license, we encrypted some code with an expiration date a while ago. Now, we need to prolong this expiration date. Is there any way we can extend it if we do not have the unencrypted code?
It would be helpful for us to prolong the expiration date or to decrypt the one with the expiration date to encrypt it again. We have the same device we used for the encryption process. 
We would appreciate any guidance or insights that can help resolve this issue. Thank you!

**Comment by jondy** [comment]
Refer to https://pyarmor.readthedocs.io/en/latest/tutorial/obfuscation.html#using-outer-file-to-store-runtime-key
I suggest first have a glance the content tree of documentation
https://pyarmor.readthedocs.io/en/latest/index.html#mastertoc

**Comment by wolf95985** [comment]
关于加密代码的到期日期，我想知道是从加密代码生成时开始计算，还是从客户第一次运行代码时开始计算。
因为有时客户拿到代码后并不会马上使用，如果是第一种情况，可能当他想到要使用时，代码已经过期了。
当然其实我也可以亲自试一下，不过更希望有官方的回答。
先行感谢！

**Comment by jondy** [comment]
到期时间是一个 epoch 值，是加密代码的时候计算的。
